### DISCUSSION TITLE
Problem with searchControl

### DISCUSSION THREAD

**Discussion by marcomile** [Leaflet/Leaflet#9310] [Q&A]
Hi all, i have this code:
$.getJSON("data/ghiacciai.geojson",function(data){
	L.geoJSON(data, {style: Stile_Ghiacciai,
		onEachFeature: function(feature, featureLayer) {
			featureLayer.bindPopup(feature.properties.Nome);
			}
		}).addTo(map);
	});

// Add search control
var searchControl = new L.Control.Search({
        position: 'topright',
        layer: $.getJSON("data/ghiacciai.geojson"), // Specify the layer to search
        propertyName: 'Nome', // Specify the property in the layer's features to search
        marker: false // Don't add markers for search results
    });
    searchControl.addTo(map);
but I can't pass the layer to the search how can I solve it? Thank you

**Comment by Falke-Design** [comment]
This should work:
```
var geojsonLayer = L.geoJSON(null, {style: Stile_Ghiacciai,
        onEachFeature: function(feature, featureLayer) {
          featureLayer.bindPopup(feature.properties.Nome);
        }
      }).addTo(map);

$.getJSON("data/ghiacciai.geojson",function(data){
      geojsonLayer.addData(data);
    });

// Add search control
    var searchControl = new L.Control.Search({
      position: 'topright',
      layer: geojsonLayer, // Specify the layer to search
      propertyName: 'Nome', // Specify the property in the layer's features to search
      marker: false // Don't add markers for search results
    });
    searchControl.addTo(map);
```
